api/src/main/java/org/openmrs/module/sync/api/db/hibernate/HibernateSyncInterceptor.java

```diff
@@ -511,28 +511,26 @@ protected void packageObject(OpenmrsObject entity, Object[] currentState, String
 				if (log.isDebugEnabled())
 					log.debug("Processing, type: " + typeName + " Field: " + propertyNames[i]);
 
-				if (propertyNames[i].equals(idPropertyName) && log.isInfoEnabled())
+				if (propertyNames[i].equals(idPropertyName) && log.isInfoEnabled()) {
 					log.debug(infoMsg + ", Id for this class: " + idPropertyName + " , value:" + currentState[i]);
+				}
 

```

Comment:
The changes here are to - instead of skipping null properties and not including them in the sync record xml (the current behavior), this includes null property values as a value in the sync record of "[null]" (this distinguishes it as well from an empty string).